src/Microsoft.Identity.Web/AuthorityHelpers.cs

```diff
@@ -51,5 +53,23 @@ internal static string EnsureAuthorityIsV2(string authority)
             preserveAuthority = true;
             return authority;
         }
+         
+        internal static void AddAuthorityQueryToOptions(MicrosoftIdentityOptions options)
+        {
+            if (options.Authority != null)
+            {
+                int queryIndex = options.Authority.IndexOf('?', StringComparison.Ordinal);
+                if (queryIndex > -1)
+                {
+                    options.ExtraQueryParameters = options.ExtraQueryParameters ?? new Dictionary<string, string>();
+                    var queryParams = HttpUtility.ParseQueryString(options.Authority[queryIndex..].TrimStart('?'));
```

Comment:
I can't find anything better with a quick search, too bad we get back a NameValueCollection  :(

AllKeys will allocate. Please avoid it. Should be able to iterate over the collection using the length and indexer. Get will still allocate an array but nothing we can do about it.

Reply:
Done, good idea to iterate rather than allocate when calling .AllKeys